src/Sarif.FunctionalTests/SpecExamples/Recommended.sarif

```diff
@@ -0,0 +1,55 @@
+{
+  "version": "1.0.0",
+  "runs": [
+    {
+      "tool": {
+        "name": "CodeScanner",
+        "semanticVersion": "2.1.0"
+      },
+      "files": {
+        "file:///user/builder/work/src/collections/list.cpp": {
+          "mimeType": "text/x-c"
+        }
+      },
+      "logicalLocations": {
+        "collections::list::add": {
+          "name": "add",
+          "kind": "function",
+          "parentKey": "collections::list"
+        },
+        "collections::list": {
+          "name": "list",
+          "kind": "type",
+          "parentKey": "collections"
+        },
+        "collections": {
+          "name": "collections",
+          "kind": "namespace"
+        }
+      },
+      "results": [
+        {
+          "ruleId": "C2001",
+          "message": "Variable \"n\" was used without being initialized.",
```

Comment:
That's correct. [`result.formattedRuleMessage`](https://rawgit.com/sarif-standard/sarif-spec/master/Static%20Analysis%20Results%20Interchange%20Format%20(SARIF).html#rule-messageformats) is an optional property ("A _result_ object **may** contain a property named `formattedRuleMessage`...). Annex G, which you haven't seen yet uses "recommended" to mean "properties that the spec says 'should' be present."